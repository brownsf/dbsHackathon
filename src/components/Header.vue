<template>
  <v-toolbar>
    <v-toolbar-title>Tech-Board</v-toolbar-title>
    <v-spacer/>
    <v-toolbar-items class="hidden-sm-and-down">
      <v-btn
        class="link"
        flat
        to="topics">Topics</v-btn>
        <v-btn
        class="link"
         v-if="loggedIn"
        flat>Logout</v-btn>
      <v-btn
        v-else
        flat
        class="link"
        to="login">Login</v-btn>
      
    </v-toolbar-items>
  </v-toolbar>
</template>

<script>
export default {
  computed: {
    loggedIn() {
      return this.$store.getters.isAuth;
    }
  }
};
</script>

