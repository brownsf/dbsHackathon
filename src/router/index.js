import Vue from 'vue';
import Router from 'vue-router';
import Register from '@/components/Register';
import Login from '@/components/Login';
import Topics from '@/components/Topics';
import Main from '@/components/Main';

Vue.use(Router);

export default new Router({
  routes: [
    {
      path: '/',
      name: 'Main',
      component: Main,
    },
    {
      path: '/register',
      name: 'Register',
      component: Register,
    },
    {
      path: '/login',
      name: 'Login',
      component: Login,
    },
    {
      path: '/topics',
      name: 'Topics',
      component: Topics,
    },
  ],
});
