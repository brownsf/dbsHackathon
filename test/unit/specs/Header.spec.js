import { mount } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vue from 'vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import Header from '@/components/Header';
import router from '@/router';

Vue.use(Vuex);
Vue.use(Vuetify);
Vue.use(VueRouter);
describe('Login', () => {
  const state = { data: {} };
  const actions = {
    register: jest.fn(),
    getAllTopics: jest.fn(),
    login: jest.fn(),
  };
  const getters = {
    isAuth: () => true,
  };

  const store = new Vuex.Store({
    state,
    actions,
    getters,
  });

  it('should render', () => {
    // expect(wrp.html()).toMatchSnapshot();
    const wrapper = mount(Header, { store, router });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should have 4 btns', () => {
    const wrapper = mount(Header, { store, router });
    const btns = wrapper.findAll('.link');
    expect(btns.length).toBe(2);
  });
});
