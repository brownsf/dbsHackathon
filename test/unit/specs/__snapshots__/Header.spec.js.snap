// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render 1`] = `
<nav class="toolbar" style="margin-top: 0px; padding-right: 0px; padding-left: 0px;">
  <div class="toolbar__content" style="height: 56px;">
    <div class="toolbar__title">Tech-Board</div>
    <div class="spacer"></div>
    <div class="toolbar__items hidden-sm-and-down">
      <a href="#/topics" class="link btn btn--flat btn--router">
        <div class="btn__content">Topics</div>
      </a>
      <button type="button" class="link btn btn--flat">
        <div class="btn__content">Logout</div>
      </button>
    </div>
  </div>
</nav>
`;
